grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -453,18 +454,34 @@ class ProductAvailabilityService {
         return quantityMap
     }
 
-    Map<Product, Map<Location, Integer>> getQuantityOnHandByProduct(Location[] locations) {
+    Map<Product, Map<Location, Integer>> getQuantityOnHandByProduct(Location[] locations, String[] categories, Boolean includeCategoryChildren) {
+        // Mysql throws an error when query with "category.id in (empty set)" is executed, but it works when NULL is included in this set
+        String queryWithCategories = categories ? "and category.id in (:categories)" : 'and category.id in (NULL)'
```

Comment:
wouldn't more reasonable be doing just a fallback to `null` if categories are empty? 
So in the `showInventoryByLocationReport` just doing:
```groovy
List categories = params.list("category").findAll { it } ?: null
```
The `findAll` is done to prevent something like this `[""]` being considered as truthy value (checkout `LocationController` 87 line): https://github.com/openboxes/openboxes/blob/63b53d3e7a9a60d4d30db09883fc0926b67cc77e/grails-app/controllers/org/pih/warehouse/core/LocationController.groovy#L86-L87